conda_rpms/tests/unit/build_rpm_structure/test_create_rpmbuild_content.py

```diff
@@ -10,97 +11,103 @@
         self.mock_create_tag = self.patch(func)
         func = 'conda_rpms.generate.render_env'
         self.mock_render_env = self.patch(func, return_value='dummy-env')
         self.config = dict(rpm=dict(prefix='prefix'))
+        self.ctag = 'env-{}-2017_01_01'.format(self.bname)
+        self.add_label(self.repo, self.branch, 'current.txt', self.ctag)
+        self.ntag = 'env-{}-2017_02_02'.format(self.bname)
+        self.add_label(self.repo, self.branch, 'next.txt', self.ntag)
+        self.count = self.branch.commit.count()
 
-    def _add_label(self, fname, tag):
-        self.branch.checkout()
-        dname = os.path.join(self.repo.working_dir, 'labels')
-        if not os.path.isdir(dname):
-            os.makedirs(dname)
-        fpath = os.path.join(dname, fname)
-        with open(fpath, 'w') as fo:
-            fo.write(tag)
-        self.repo.index.add([fpath])
-        comment = 'Add label {}'.format(fname)
-        self.repo.index.commit(comment)
+    def tearDown(self):
+        if os.path.exists(self.repo.working_dir):
+            shutil.rmtree(self.repo.working_dir)
 
-    def _check_full_build(self, dname, state, ctag, ntag, count):
+    def _check_full_build(self, dname, state):
         create_rpmbuild_content(self.repo, dname, self.config, state)
         self.assertEqual(self.mock_create_tag.call_count, 2)
-        expected = [((self.repo, ctag, dname, self.config),),
-                    ((self.repo, ntag, dname, self.config),)]
+        expected = [((self.repo, self.ctag, dname, self.config),),
+                    ((self.repo, self.ntag, dname, self.config),)]
         self.assertEqual(self.mock_create_tag.call_args_list, expected)
         self.assertEqual(self.mock_render_env.call_count, 2)
-        expected = [((self.bname, 'current', self.config, ctag, count),),
-                    ((self.bname, 'next', self.config, ntag, count),)]
+        expected = [((self.bname, 'current', self.config, self.ctag,
+                      self.count),),
+                    ((self.bname, 'next', self.config, self.ntag,
+                      self.count),)]
         self.assertEqual(self.mock_render_env.call_args_list, expected)
-        self.mock_create_tag.reset_mock()
-        self.mock_render_env.reset_mock()
-
-    def test_basic(self):
-        ctag = 'env-{}-2017_01_01'.format(self.bname)
-        self._add_label('current.txt', ctag)
-        ntag = 'env-{}-2017_02_02'.format(self.bname)
-        self._add_label('next.txt', ntag)
-        count = self.branch.commit.count()
 
+    def test_build_all_default(self):
         with self.temp_dir() as dname:
             # Make the expected RPM build SPECS directory.
             os.makedirs(os.path.join(dname, 'SPECS'))
-
             # Test - no prior build, so both labels build.
             state = {}
-            self._check_full_build(dname, state, ctag, ntag, count)
+            self._check_full_build(dname, state)
 
+    def test_build_all_with_mismatch(self):
```

Comment:
Could you rename this so that it's clear that "mismatch" refers to the tags, something like `test_build_all_with_mismatch_tags` or even `test_build_all_branches_with_mismatch_tags`